### PR TITLE
Ihasan/choose bins

### PR DESCRIPTION
**Pull request by ih64**
this branch is for #25 aimed at the first bullet point. Not yet ready to merge, but ready for comments. 

atm there is a config option for lens and source bin indices. If they are both left as empty lists, all possible bin pairs are run. if they are both lists with specified indices, then combinations of those bins are ran. Probably a neater way of handling this. Still need to add an option to handle when the user only specifies lens bins or only source bins.   

still checking if the changes made to write_output() work with sacc format, too.

### CODE REVIEW

File: txpipe/twopoint.py [hunk 1]
```diff
@@ -55,20 +57,28 @@ def run(self):
         if self.comm:
             self.comm.Barrier()
 
-        # Get the number of bins from the
-        # tomography input file
-        nbins_source, nbins_lens  = self.read_nbins()
+        # the default is to run all bins, which is set in the config_options above
+        if self.config['source_bins'][0] == -1 and self.config['lens_bins'][0] == -1:
```
Comment (joezuntz) [inline txpipe/twopoint.py]: We might also want to deal with the case where one of the sets of bins is empty, but I see you've put that as a TODO later on so we can deal with it later.

**Comment by joezuntz** [comment]
Just seen that there are some conflicts here - could you (@ih64 ) have a look at them?

File: txpipe/twopoint.py (hunk 1 above)
Comment (ih64) [reply txpipe/twopoint.py]: sounds good, I can keep working on this development branch to include that feature

**Comment by ih64** [comment]
thanks for the review! the merge conflict was a harmless one, it looks like there were other changes made to the config to add in reduce_random_size in addition to the lens and source bins list i added.

**Pull request merged**